File: AtomykPlay/app/src/main/java/com/atomykcoder/atomykplay/repository/MusicRepository.java
package com.atomykcoder.atomykplay.repository;

import android.content.Context;

import com.atomykcoder.atomykplay.data.Music;
import com.atomykcoder.atomykplay.interfaces.MusicDaoI;

import java.util.ArrayList;
import java.util.concurrent.CompletableFuture;

public class MusicRepository {
    private final MusicDaoI musicDaoI;
    private final Context context;

    public MusicRepository(Context context) {
        musicDaoI = new MusicDaoImpl();
        this.context = context;
    }

    public CompletableFuture<ArrayList<Music>> fetchMusicList() {
        CompletableFuture<ArrayList<Music>> future = new CompletableFuture<>();
        musicDaoI.fetchMusic(context).thenAccept(future::complete).exceptionally(it -> {
            future.completeExceptionally(it);
            return null;
        });
        return future;
    }
}

File: AtomykPlay/app/src/main/java/com/atomykcoder/atomykplay/repository/MusicRepository.kt
package com.atomykcoder.atomykplay.repository

import android.content.Context
import com.atomykcoder.atomykplay.interfaces.MusicDaoI
import com.atomykcoder.atomykplay.models.Music
import java.util.concurrent.CompletableFuture

class MusicRepository(context: Context) {
    private val musicDaoI: MusicDaoI
    private val context: Context

    init {
        musicDaoI = MusicDaoImpl()
        this.context = context
    }

    fun fetchMusicList(): CompletableFuture<ArrayList<Music>> {
        val future = CompletableFuture<ArrayList<Music>>()
        musicDaoI.fetchMusic(context)
            .thenAccept { value: ArrayList<Music> -> future.complete(value) }
            .exceptionally {
                future.completeExceptionally(it)
                null
            }
        return future
    }
}